src/python/pants/backend/python/target_types_rules.py

```diff
@@ -316,7 +316,12 @@ async def resolve_python_distribution_entry_points(
     # Intermediate step, as Get(Targets) returns a deduplicated set.. which breaks in case of
     # multiple input refs that maps to the same target.
     target_addresses = await Get(
-        Addresses, UnparsedAddressInputs(target_refs, owning_address=address)
+        Addresses,
+        UnparsedAddressInputs(
+            target_refs,
+            owning_address=address,
+            description_of_origin=f"TODO(#14468)",
```

Comment:
I can figure this one out after lunch. The tricky part is it could be from `provides` field or `entry_points` field.